src-tauri/src/permissions.rs

```diff
@@ -0,0 +1,65 @@
+use Permission::*;
+use serde_json::{to_string_pretty, Value};
+use std::{fs::File, io::{Read, Write}};
+
+#[derive(Clone)]
+pub enum Permission {
+    Clipboard,
+    Location,
+    Microphone,
+    Screenshot,
+    Tts
+}
+
+impl Permission {
+    pub fn as_str(&self) -> &str {
+        match *self {
+            Clipboard => "Clipboard",
+            Location => "Location",
+            Microphone => "Microphone",
+            Screenshot => "Screenshot",
+            Tts => "Tts"
+        }
+    }
+}
+
+pub fn update(permissions: Value) {
+    let pretty_json = to_string_pretty(&permissions).unwrap();
+    let mut file = File::create("permissions.json").expect("Failed to create permissions.json!");
```

Comment:
https://doc.rust-lang.org/std/fs/struct.File.html#method.create
apparently it "truncates" it, which doesn't sound right. I think what we actually need is:
https://doc.rust-lang.org/std/fs/fn.write.html#

Reply:
but actually since we are using write_all on line 29 I think it accomplishes the same as fs::write. I would like to change it to fs::write anyway for clean sake